### PR TITLE
ES2017 review third round

### PR DESCRIPTION
**Pull request by anba**
Maybe it's time to update https://tc39.github.io/ecma262/#sec-algorithm-conventions to allow the `?` short-hand when invoking algorithms associated to grammar productions instead of just abstract operations.

### CODE REVIEW

**Comment by bterlson** [comment]
I think it is, I didn't notice that fc1cbe3 depended on that. I wasn't too happy with the `?` operator just kind of sitting after perform but allowing the convention and keeping fc1cbe3 is my preferred approach. What do you think?

**Comment by anba** [comment]
I think I'd probably revert fc1cbe3e19798e2611d3947abd77e4cdfe64f1f9 for now, and when https://tc39.github.io/ecma262/#sec-algorithm-conventions was updated, we can re-apply fc1cbe3e19798e2611d3947abd77e4cdfe64f1f9 and even get rid of many more ReturnIfAbrupt calls in the spec.

File: spec.html [hunk 1]
```diff
@@ -9543,10 +9542,10 @@
         &lt;ZWJ&gt;
 
       UnicodeIDStart ::
-        &gt; any Unicode code point with the Unicode property &ldquo;ID_Start&rdquo;
+        &gt; any Unicode code point with the Unicode property “ID_Start”
```
Comment (mathiasbynens) [inline spec.html]: These would be the first non-ASCII symbols in `spec.html`. Are we sure this doesn’t break anyone’s scripts? (It shouldn’t, but hey…)

FWIW, here’s the ecmarkup bug: https://github.com/bterlson/ecmarkup/issues/118. Fixing that would be preferable to this workaround IMHO.

Comment (anba) [inline spec.html]: Agreed, I didn't know this was a known issue in ecmarkup. Looks like there's even an open PR to fix this bug: https://github.com/bterlson/ecmarkup/pull/117

**Comment by bterlson** [comment]
I've committed most of these (minor editorial fixes also committed to es2017, sending final draft to ECMA tomorrow). There are a few I've left off:

* 685a892: I actually always intended that *'s be used for primitive values, but I see ` has taken off. Should we get rid of the distinction and also use ` for numbers and Booleans? /cc @jmdyck 
* 1e37b9b: There are more instances of this (and presumably non-strict code code should be non-strict mode code?) I'd rather leave it for now until I find where I previously discussed this.

**Comment by jmdyck** [comment]
The MS Word version of the spec presented string literals pretty consistently in monospace font, so those all became backticks in ecmarkdown. (So it hasn't really "taken off", it's always been there.)

It would probably be more consistent with the current definition of ecmarkdown to use asterisks for string values, as you intended (except if/where the spec is specifically talking about a string literal as a chunk of source code, which is rare, I think).

The opposite change (using backtick for numbers and booleans) would, I think, obscure the distinction between (concrete) source code and (abstract) values.

**Comment by domenic** [comment]
My preference would be stick with backticks for strings and no-special-font for number literals or other primitives, just for the smallest delta from the current spec. Then figure out our story for number literals per https://github.com/tc39/proposal-bigint/issues/10, later.

**Comment by jmdyck** [comment]
For some previous discussion of "[non-]strict mode code" vs "[non-]strict code", see https://bugs.ecmascript.org/show_bug.cgi?id=4144 and https://bugs.ecmascript.org/show_bug.cgi?id=4324

**Comment by bterlson** [comment]
@jmdyck Indeed you are correct, I've used *'s in a few other specs but never ecma262 (except by mistake).

@Domenic I agree we should clean up number notations and then make changes.

**Pull request closed without merging**